iothub/service/tests/DeserializationTests.cs

```diff
@@ -0,0 +1,303 @@
+            clientTwin.DeviceId.Should().Be(ExpectedDeviceId);
+        }
+
+        [TestMethod]
+        public void Configuration_TestWithSchema_Ok()
+        {
+            const string ExpectedDeviceId = "aa";
+            const string ExpectedSchemaVersion = "1.0";
+            const string JsonString = @"
+{
+  ""id"": ""aa"",
+  ""schemaVersion"": ""1.0"",
+  ""content"": {
+    ""modulesContent"": {
+        ""$edgeAgent"": {
+            ""properties.desired"": {
+                ""schemaVersion"": ""1.0""
+            }
+        }
+    }
+  }
+}";
+
+            Configuration configuration = JsonConvert.DeserializeObject<Configuration>(JsonString);
+            configuration.Id.Should().Be(ExpectedDeviceId);
+            configuration.SchemaVersion.Should().Be(ExpectedSchemaVersion);
+        }
+
+        [TestMethod]
+        public void Configuration_TestNullSchema_Ok()
+        {
+            const string ExpectedDeviceId = "aa";
+            const string ExpectedSchemaVersion = "1.0";
+            const string JsonString = @"
+{
+  ""id"": ""aa"",
+  ""content"": {
+    ""modulesContent"": {
+        ""$edgeAgent"": {
+            ""properties.desired"": {
+            }
+        }
+    }
+  }
+}";
+            Configuration configuration = JsonConvert.DeserializeObject<Configuration>(JsonString);
+            configuration.Id.Should().Be(ExpectedDeviceId);
+            configuration.SchemaVersion.Should().Be(ExpectedSchemaVersion);
+        }
+
+        [TestMethod]
+        public void ImportConfiguration_Deserialize_Ok()
+        {
+            const string ExpectedDeviceId = "aa";
+            const string ExpectedSchemaVersion = "1.0";
+            const string JsonString = @"
+{
+  ""id"": ""aa"",
+  ""schemaVersion"": ""1.0"",
+  ""importMode"": ""CreateOrUpdateIfMatchETag"",
+  ""content"": {
+    ""modulesContent"": {
+        ""$edgeAgent"": {
+            ""properties.desired"": {
+            }
+        }
+    }
+  }
+}";
+            ImportConfiguration importConfiguration = JsonConvert.DeserializeObject<ImportConfiguration>(JsonString);
+            importConfiguration.Id.Should().Be(ExpectedDeviceId);
+            importConfiguration.SchemaVersion.Should().Be(ExpectedSchemaVersion);
+            importConfiguration.ImportMode.Should().Be(ConfigurationImportMode.CreateOrUpdateIfMatchETag);
+        }
+
+        [TestMethod]
+        public void FeedbackRecord_Deserialize_Ok()
+        {
+            const string ExpectedOriginalMessageId = "1";
+            const string ExpectedDeviceMessageId = "2";
+            const string ExpectedDeviceId = "testDeviceId";
+            const string ExpectedDescription = "Success";
+            var expectedEnqueuedTimeUtc = new DateTimeOffset(2023, 1, 20, 8, 6, 32,
+                                                new TimeSpan(1, 0, 0));
+            const string JsonString = @"
+{
+  ""originalMessageId"": 1,
```

Comment:
If you are going to have constants above (which I encourage), I'd recommend using that to create this string. That way you don't have 2 places the value is specified.

Reply:
But my earlier comment about creating an object and then serializing and deserializing would supersede this.